### ISSUE TITLE
Frequent crashes on macOS 13.0 Ventura

### ISSUE THREAD

**Issue by NeoVG** [riok/Kreya#111]
**Describe the bug**

Since upgrading macOS to the newly released version 13.0, Kreya crashes directly on startup or shortly after.

The error is always the same type (EXC_BAD_ACCESS (SIGSEGV)) and code (KERN_INVALID_ADDRESS) and happens somewhere inside the JavaScriptEngine of WebKit, so I assume it's caused by an incompatibility between Kreya and the WebKit libs shipped with the new version of macOS.

Happens with both the ARM binary as well as the x86 binary running via Rosetta.

**To Reproduce**

Start the app, load a project, do a request, and it's probably crashed already.

**Expected behavior**

App doesn't crash all the time. ;)

**Error Report**

```-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               Kreya.app [48461]
Path:                  /Applications/Kreya.app/Contents/MacOS/Kreya.app
Identifier:            app.kreya
Version:               1.8.1 (1.8.1)
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2022-10-25 18:24:49.4715 +0200
OS Version:            macOS 13.0 (22A380)
Report Version:        12
Anonymous UUID:        0FA69590-A8E1-4054-B71E-5924EBBE665F


Time Awake Since Boot: 27000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000002
Exception Codes:       0x0000000000000001, 0x0000000000000002

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [48461]

VM Region Info: 0x2 is not in any region.  Bytes before following region: 68719476734
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      commpage (reserved)        1000000000-7000000000   [384.0G] ---/--- SM=NUL  ...(unallocated)

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libsystem_blocks.dylib        	       0x188d63bc8 HelperBase<GenericOutOfLine>::destroyBlock(Block_layout*, bool, unsigned char*) + 68
1   libsystem_blocks.dylib        	       0x188d631a0 _call_dispose_helpers_excp + 72
2   libsystem_blocks.dylib        	       0x188d631a0 _call_dispose_helpers_excp + 72
3   libsystem_blocks.dylib        	       0x188d62f54 _Block_release + 252
4   WebKit                        	       0x1a88248a8 WTF::Detail::CallableWrapper<-[WKWebView _evaluateJavaScript:asAsyncFunction:withSourceURL:withArguments:forceUserGesture:inFrame:inWorld:completionHandler:]::$_14, void, std::experimental::fundamentals_v3::expected<WTF::RefPtr<API::SerializedScriptValue, WTF::RawPtrTraits<API::SerializedScriptValue>, WTF::DefaultRefDerefTraits<API::SerializedScriptValue> >, WebCore::ExceptionDetails>&&>::~CallableWrapper() + 92
5   WebKit                        	       0x1a8b1aadc WTF::Detail::CallableWrapper<WebKit::WebPageProxy::runJavaScriptInFrameInScriptWorld(WebCore::RunJavaScriptParameters&&, std::__1::optional<WTF::ObjectIdentifier<WebCore::FrameIdentifierType> >, API::ContentWorld&, WTF::CompletionHandler<void (std::experimental::fundamentals_v3::expected<WTF::RefPtr<API::SerializedScriptValue, WTF::RawPtrTraits<API::SerializedScriptValue>, WTF::DefaultRefDerefTraits<API::SerializedScriptValue> >, WebCore::ExceptionDetails>&&)>&&)::$_25, void, WTF::Span<unsigned char const, 18446744073709551615ul>&&, std::__1::optional<WebCore::ExceptionDetails>&&>::call(WTF::Span<unsigned char const, 18446744073709551615ul>&&, std::__1::optional<WebCore::ExceptionDetails>&&) + 376
6   WebKit                        	       0x1a8e64aa8 Messages::WebPage::RunJavaScriptInFrameInScriptWorld::callReply(IPC::Decoder&, WTF::CompletionHandler<void (WTF::Span<unsigned char const, 18446744073709551615ul>&&, std::__1::optional<WebCore::ExceptionDetails>&&)>&&) + 148
7   WebKit                        	       0x1a8b1a718 WTF::Detail::CallableWrapper<unsigned long long IPC::MessageSender::sendWithAsyncReply<Messages::WebPage::RunJavaScriptInFrameInScriptWorld, WebKit::WebPageProxy::runJavaScriptInFrameInScriptWorld(WebCore::RunJavaScriptParameters&&, std::__1::optional<WTF::ObjectIdentifier<WebCore::FrameIdentifierType> >, API::ContentWorld&, WTF::CompletionHandler<void (std::experimental::fundamentals_v3::expected<WTF::RefPtr<API::SerializedScriptValue, WTF::RawPtrTraits<API::SerializedScriptValue>, WTF::DefaultRefDerefTraits<API::SerializedScriptValue> >, WebCore::ExceptionDetails>&&)>&&)::$_25>(Messages::WebPage::RunJavaScriptInFrameInScriptWorld&&, WebKit::WebPageProxy::runJavaScriptInFrameInScriptWorld(WebCore::RunJavaScriptParameters&&, std::__1::optional<WTF::ObjectIdentifier<WebCore::FrameIdentifierType> >, API::ContentWorld&, WTF::CompletionHandler<void (std::experimental::fundamentals_v3::expected<WTF::RefPtr<API::SerializedScriptValue, WTF::RawPtrTraits<API::SerializedScriptValue>, WTF::DefaultRefDerefTraits<API::SerializedScriptValue> >, WebCore::ExceptionDetails>&&)>&&)::$_25&&, unsigned long long, WTF::OptionSet<IPC::SendOption>)::'lambda'(IPC::Decoder*), void, IPC::Decoder*>::call(IPC::Decoder*) + 60
8   WebKit                        	       0x1a8a61450 WTF::Detail::CallableWrapper<WebKit::AuxiliaryProcessProxy::sendMessage(WTF::UniqueRef<IPC::Encoder>&&, WTF::OptionSet<IPC::SendOption>, std::__1::optional<std::__1::pair<WTF::CompletionHandler<void (IPC::Decoder*)>, unsigned long long> >&&, WebKit::AuxiliaryProcessProxy::ShouldStartProcessThrottlerActivity)::$_1, void, IPC::Decoder*>::call(IPC::Decoder*) + 64
9   WebKit                        	       0x1a88e0a10 IPC::Connection::dispatchMessage(std::__1::unique_ptr<IPC::Decoder, std::__1::default_delete<IPC::Decoder> >) + 1004
10  WebKit                        	       0x1a88dffb0 IPC::Connection::dispatchIncomingMessages() + 268
11  JavaScriptCore                	       0x1a21cc8f8 WTF::RunLoop::performWork() + 200
12  JavaScriptCore                	       0x1a21cd630 WTF::RunLoop::performWork(void*) + 36
13  CoreFoundation                	       0x1890e5a34 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
14  CoreFoundation                	       0x1890e59c8 __CFRunLoopDoSource0 + 176
15  CoreFoundation                	       0x1890e5738 __CFRunLoopDoSources0 + 244
16  CoreFoundation                	       0x1890e433c __CFRunLoopRun + 836
17  CoreFoundation                	       0x1890e38a4 CFRunLoopRunSpecific + 612
18  HIToolbox                     	       0x1927573bc RunCurrentEventLoopInMode + 292
19  HIToolbox                     	       0x192757200 ReceiveNextEventCommon + 672
20  HIToolbox                     	       0x192756f48 _BlockUntilNextEventMatchingListInModeWithFilter + 72
21  AppKit                        	       0x18c33c630 _DPSNextEvent + 632
22  AppKit                        	       0x18c33b7c0 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 728
23  AppKit                        	       0x18c32fbf0 -[NSApplication run] + 464
24  ???                           	       0x2815b9354 ???
25  ???                           	       0x2815b9658 ???
26  ???                           	       0x282210760 ???
27  ???                           	       0x28221070c ???
28  ???                           	       0x28220f55c ???
29  ???                           	       0x281586a8c ???
30  ???                           	       0x280e69610 ???
31  Kreya.app                     	       0x100ba6688 CallDescrWorkerInternal + 132
32  Kreya.app                     	       0x100a170ec MethodDescCallSite::CallTargetWorker(unsigned long const*, unsigned long*, int) + 868
33  Kreya.app                     	       0x10090ec04 RunMain(MethodDesc*, short, int*, PtrArray**) + 628
34  Kreya.app                     	       0x10090ef08 Assembly::ExecuteMainMethod(PtrArray**, int) + 376
35  Kreya.app                     	       0x10093d278 CorHost2::ExecuteAssembly(unsigned int, char16_t const*, int, char16_t const**, unsigned int*) + 476
36  Kreya.app                     	       0x100e130f4 coreclr_execute_assembly + 208
37  Kreya.app                     	       0x1008d8c44 run_app_for_context(hostpolicy_context_t const&, int, char const**) + 1056
38  Kreya.app                     	       0x1008d9984 corehost_main + 240
39  Kreya.app                     	       0x1008ba720 fx_muxer_t::handle_exec_host_command(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, host_startup_info_t const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::unordered_map<known_options, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >, known_options_hash, std::__1::equal_to<known_options>, std::__1::allocator<std::__1::pair<known_options const, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > > > > const&, int, char const**, int, host_mode_t, bool, char*, int, int*) + 1320
40  Kreya.app                     	       0x1008b980c fx_muxer_t::execute(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int, char const**, host_startup_info_t const&, char*, int, int*) + 856
41  Kreya.app                     	       0x1008b62ec hostfxr_main_bundle_startupinfo + 196
42  Kreya.app                     	       0x1008f9768 exe_start(int, char const**) + 1124
43  Kreya.app                     	       0x1008f9a58 main + 152
44  dyld                          	       0x188cdbe50 start + 2544
```

**Environment (if possible, copy the information from the error dialog or the About menu):**

 - OS: macOS 13.0
 - Kreya Version: 1.8.1 and 1.9.0-alpha.1 (same problem with both versions)

**Comment by CommonGuy** [comment]
Thanks for you detailed bug report. We haven't tested Kreya on macOS 13 yet... We will look into this

**Comment by CommonGuy** [comment]
We just release 1.9.0-alpha.2 with a fix for macOS. Could you try that and report back whether that fixes your issue?

**Comment by NeoVG** [comment]
> We just released 1.9.0-alpha.2 with a fix for macOS. Could you try that and report back whether that fixes your issue?

Thanks for the quick response! <3

Unfortunately the problem remains and the app crashes just as fast. :(

Luckily I found a workaround by just installing macOS 12.6 in a UTM VM and starting Kreya there, which works perfectly well for now. :) Would have been hard otherwise, as Kreya seems to be the only really usable gRPC client that also supports OAuth (which is a must for my use case). So thanks for the work and keep it up! :)

New error report:

```-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               Kreya.app [35347]
Path:                  /Applications/Kreya.app/Contents/MacOS/Kreya.app
Identifier:            app.kreya
Version:               1.9.0-alpha.2 (1.9.0-alpha.2)
Code Type:             ARM-64 (Native)
Parent Process:        launchd [1]
User ID:               501

Date/Time:             2022-10-27 16:08:33.4716 +0200
OS Version:            macOS 13.0 (22A380)
Report Version:        12
Anonymous UUID:        0FA69590-A8E1-4054-B71E-5924EBBE665F


Time Awake Since Boot: 190000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000001
Exception Codes:       0x0000000000000001, 0x0000000000000001

Termination Reason:    Namespace SIGNAL, Code 11 Segmentation fault: 11
Terminating Process:   exc handler [35347]

VM Region Info: 0x1 is not in any region.  Bytes before following region: 68719476735
      REGION TYPE                    START - END         [ VSIZE] PRT/MAX SHRMOD  REGION DETAIL
      UNUSED SPACE AT START
--->  
      commpage (reserved)        1000000000-7000000000   [384.0G] ---/--- SM=NUL  ...(unallocated)

Thread 0 Crashed::  Dispatch queue: com.apple.main-thread
0   libsystem_blocks.dylib        	       0x188d63bc8 HelperBase<GenericOutOfLine>::destroyBlock(Block_layout*, bool, unsigned char*) + 68
1   libsystem_blocks.dylib        	       0x188d631a0 _call_dispose_helpers_excp + 72
2   libsystem_blocks.dylib        	       0x188d631a0 _call_dispose_helpers_excp + 72
3   libsystem_blocks.dylib        	       0x188d62f54 _Block_release + 252
4   WebKit                        	       0x1a88248a8 WTF::Detail::CallableWrapper<-[WKWebView _evaluateJavaScript:asAsyncFunction:withSourceURL:withArguments:forceUserGesture:inFrame:inWorld:completionHandler:]::$_14, void, std::experimental::fundamentals_v3::expected<WTF::RefPtr<API::SerializedScriptValue, WTF::RawPtrTraits<API::SerializedScriptValue>, WTF::DefaultRefDerefTraits<API::SerializedScriptValue> >, WebCore::ExceptionDetails>&&>::~CallableWrapper() + 92
5   WebKit                        	       0x1a8b1aadc WTF::Detail::CallableWrapper<WebKit::WebPageProxy::runJavaScriptInFrameInScriptWorld(WebCore::RunJavaScriptParameters&&, std::__1::optional<WTF::ObjectIdentifier<WebCore::FrameIdentifierType> >, API::ContentWorld&, WTF::CompletionHandler<void (std::experimental::fundamentals_v3::expected<WTF::RefPtr<API::SerializedScriptValue, WTF::RawPtrTraits<API::SerializedScriptValue>, WTF::DefaultRefDerefTraits<API::SerializedScriptValue> >, WebCore::ExceptionDetails>&&)>&&)::$_25, void, WTF::Span<unsigned char const, 18446744073709551615ul>&&, std::__1::optional<WebCore::ExceptionDetails>&&>::call(WTF::Span<unsigned char const, 18446744073709551615ul>&&, std::__1::optional<WebCore::ExceptionDetails>&&) + 376
6   WebKit                        	       0x1a8e64aa8 Messages::WebPage::RunJavaScriptInFrameInScriptWorld::callReply(IPC::Decoder&, WTF::CompletionHandler<void (WTF::Span<unsigned char const, 18446744073709551615ul>&&, std::__1::optional<WebCore::ExceptionDetails>&&)>&&) + 148
7   WebKit                        	       0x1a8b1a718 WTF::Detail::CallableWrapper<unsigned long long IPC::MessageSender::sendWithAsyncReply<Messages::WebPage::RunJavaScriptInFrameInScriptWorld, WebKit::WebPageProxy::runJavaScriptInFrameInScriptWorld(WebCore::RunJavaScriptParameters&&, std::__1::optional<WTF::ObjectIdentifier<WebCore::FrameIdentifierType> >, API::ContentWorld&, WTF::CompletionHandler<void (std::experimental::fundamentals_v3::expected<WTF::RefPtr<API::SerializedScriptValue, WTF::RawPtrTraits<API::SerializedScriptValue>, WTF::DefaultRefDerefTraits<API::SerializedScriptValue> >, WebCore::ExceptionDetails>&&)>&&)::$_25>(Messages::WebPage::RunJavaScriptInFrameInScriptWorld&&, WebKit::WebPageProxy::runJavaScriptInFrameInScriptWorld(WebCore::RunJavaScriptParameters&&, std::__1::optional<WTF::ObjectIdentifier<WebCore::FrameIdentifierType> >, API::ContentWorld&, WTF::CompletionHandler<void (std::experimental::fundamentals_v3::expected<WTF::RefPtr<API::SerializedScriptValue, WTF::RawPtrTraits<API::SerializedScriptValue>, WTF::DefaultRefDerefTraits<API::SerializedScriptValue> >, WebCore::ExceptionDetails>&&)>&&)::$_25&&, unsigned long long, WTF::OptionSet<IPC::SendOption>)::'lambda'(IPC::Decoder*), void, IPC::Decoder*>::call(IPC::Decoder*) + 60
8   WebKit                        	       0x1a8a61450 WTF::Detail::CallableWrapper<WebKit::AuxiliaryProcessProxy::sendMessage(WTF::UniqueRef<IPC::Encoder>&&, WTF::OptionSet<IPC::SendOption>, std::__1::optional<std::__1::pair<WTF::CompletionHandler<void (IPC::Decoder*)>, unsigned long long> >&&, WebKit::AuxiliaryProcessProxy::ShouldStartProcessThrottlerActivity)::$_1, void, IPC::Decoder*>::call(IPC::Decoder*) + 64
9   WebKit                        	       0x1a88e0a10 IPC::Connection::dispatchMessage(std::__1::unique_ptr<IPC::Decoder, std::__1::default_delete<IPC::Decoder> >) + 1004
10  WebKit                        	       0x1a88dffb0 IPC::Connection::dispatchIncomingMessages() + 268
11  JavaScriptCore                	       0x1a21cc8f8 WTF::RunLoop::performWork() + 200
12  JavaScriptCore                	       0x1a21cd630 WTF::RunLoop::performWork(void*) + 36
13  CoreFoundation                	       0x1890e5a34 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 28
14  CoreFoundation                	       0x1890e59c8 __CFRunLoopDoSource0 + 176
15  CoreFoundation                	       0x1890e5738 __CFRunLoopDoSources0 + 244
16  CoreFoundation                	       0x1890e433c __CFRunLoopRun + 836
17  CoreFoundation                	       0x1890e38a4 CFRunLoopRunSpecific + 612
18  HIToolbox                     	       0x1927573bc RunCurrentEventLoopInMode + 292
19  HIToolbox                     	       0x192757200 ReceiveNextEventCommon + 672
20  HIToolbox                     	       0x192756f48 _BlockUntilNextEventMatchingListInModeWithFilter + 72
21  AppKit                        	       0x18c33c630 _DPSNextEvent + 632
22  AppKit                        	       0x18c33b7c0 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 728
23  AppKit                        	       0x18c32fbf0 -[NSApplication run] + 464
24  ???                           	       0x281449154 ???
25  ???                           	       0x281449458 ???
26  ???                           	       0x2820f1688 ???
27  ???                           	       0x2820f1634 ???
28  ???                           	       0x2820eaab4 ???
29  ???                           	       0x281414de4 ???
30  ???                           	       0x280df9ef0 ???
31  Kreya.app                     	       0x10292e688 CallDescrWorkerInternal + 132
32  Kreya.app                     	       0x10279f0ec MethodDescCallSite::CallTargetWorker(unsigned long const*, unsigned long*, int) + 868
33  Kreya.app                     	       0x102696c04 RunMain(MethodDesc*, short, int*, PtrArray**) + 628
34  Kreya.app                     	       0x102696f08 Assembly::ExecuteMainMethod(PtrArray**, int) + 376
35  Kreya.app                     	       0x1026c5278 CorHost2::ExecuteAssembly(unsigned int, char16_t const*, int, char16_t const**, unsigned int*) + 476
36  Kreya.app                     	       0x102b9b0f4 coreclr_execute_assembly + 208
37  Kreya.app                     	       0x102660c44 run_app_for_context(hostpolicy_context_t const&, int, char const**) + 1056
38  Kreya.app                     	       0x102661984 corehost_main + 240
39  Kreya.app                     	       0x102642720 fx_muxer_t::handle_exec_host_command(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, host_startup_info_t const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::unordered_map<known_options, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >, known_options_hash, std::__1::equal_to<known_options>, std::__1::allocator<std::__1::pair<known_options const, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > > > > const&, int, char const**, int, host_mode_t, bool, char*, int, int*) + 1320
40  Kreya.app                     	       0x10264180c fx_muxer_t::execute(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, int, char const**, host_startup_info_t const&, char*, int, int*) + 856
41  Kreya.app                     	       0x10263e2ec hostfxr_main_bundle_startupinfo + 196
42  Kreya.app                     	       0x102681768 exe_start(int, char const**) + 1124
43  Kreya.app                     	       0x102681a58 main + 152
44  dyld                          	       0x188cdbe50 start + 2544```

**Comment by arcilli** [comment]
Hello! Just reproduced this again on M1 Pro with macOS 13.0 Ventura.

Is someone looking at it?

Also, from where could I download 1.9.0-alpha.2?

**Comment by CommonGuy** [comment]
We are looking into this.

1.9.0-alpha.2 can be downloaded from https://alpha-downloads.kreya.app/Kreya-macos-arm64.zip (or by switching the release channel in Kreya -> About), but it seems like it won't fix the issue.

**Comment by CommonGuy** [comment]
We just released 1.9.0-alpha.3, which can be downloaded from https://alpha-downloads.kreya.app/Kreya-macos-arm64.zip. The release was tested on macOS 13.0. Could you test whether this release works for you as well?

**Comment by pnshiralkar** [comment]
Tested it on MacOS 13.0 (Macbook Pro 2021, M1 pro)
Was facing constant crashes on current stable release.
Did basic operations, seems to be working fine now on 1.9.0-alpha.3. We may need to wait for some time to see if any other issue occurs.
@CommonGuy Thanks a lot for this quick fix!

**Comment by CommonGuy** [comment]
We just released Kreya 1.9 as a stable release, which contains the bugfix for macOS 13.

Users which installed an alpha release may switch to the stable release channel: In the Kreya menu, click Kreya -> About, switch the channel from Alpha to Stable and search for updates.
